src/module/combat/websocket/ws-combat/user/WSUserCombatManager.ts

```diff
@@ -71,55 +72,34 @@ export class WSUserCombatManager extends WSCombatManager<ABFWSUserRequest, ABFWS
     return this.game.user!;
   }
 
-  get token(): TokenDocument {
-    if (!this.user.character?.id) {
-      const message = this.game.i18n.localize('macros.combat.dialog.error.noSelectedActor.title');
-      ABFDialogs.prompt(message);
-
-      throw new Error(message);
-    }
-
-    const token = this.game.scenes?.current?.tokens.find(t => t.data.actorId === this.user.character!.id);
-
-    if (!token?.id) {
-      const message = this.game.i18n.localize('macros.combat.dialog.error.noTokenInCurrentScene.title');
-      ABFDialogs.prompt(message);
-
-      throw new Error(message);
-    }
-
-    return token;
-  }
-
-  private isMyToken(tokenId): boolean {
-    return this.token.id === tokenId;
+  private isMyToken(tokenId: string): boolean {
+    return this.game.canvas.tokens?.ownedTokens.filter(tk => tk.id === tokenId).length === 1;
```

Comment:
@Eryx5502 

Yo restringiría el uso de los tokens a lo que hay dentro de la escena actual, ahora mismo el combate es así por no acomplejizarlo.

```suggestion
    return this.game.scenes?.current?.tokens?.ownedTokens.filter(tk => tk.id === tokenId).length === 1;
```

Reply:
La documentación no es la mejor, pero creo que `game.scenes.current.tokens` es simplemente un `Map` de los tokens y no tiene `ownedTokens`. Creo que `game.canvas.tokens` es un objeto especial que definen ellos y que los tokens que están ahí son los que están en el momento en el "canvas" así que serían los que se están mostrando en la escena en la que está el usuario.

Estuve mirando la documentación y me pareció que era la única manera de evitar usar la propiedad `token._controlled`, que es privada, directamente. Aunque si pensáis que es mejor podemos hacer algo parecido a lo que estaba para usar `_controlled`.